addons/common/functions/fnc_getPylonTurret.sqf

```diff
@@ -0,0 +1,64 @@
+/*
+ * Author: PabstMirror
+ * Finds turret owner of a pylon
+ *
+ * Arguments:
+ * 0: Vehicle <OBJECT>
+ * 1: Pylon Index (starting at 0) <NUMBER>
+ *
+ * Return Value:
+ * <ARRAY> (turret Index, either [-1] or [0])
+ *
+ * Example:
+ * [cursorObject, 0] call ace_common_fnc_getPylonTurret
+ *
+ * Public: No
+ */
+// #define DEBUG_MODE_FULL
+#include "script_component.hpp"
+
+params ["_vehicle", "_pylonIndex"];
+
+// See if index is in ace_pylonTurrets setVar on vehicle
+private _pylonTurrets = _vehicle getVariable ["ace_pylonTurrets", []];
+private _returnValue = _pylonTurrets param [_pylonIndex, []];
+
+if (!(_returnValue isEqualTo [])) then {
+    TRACE_1("Using ace_pylonTurrets value",_savedValue);
+} else {
+    // Attempt to determine turret owner based on magazines in the vehicle
+    private _pyMags = getPylonMagazines _vehicle;
+    private _pylonConfigs = configProperties [configFile >> "CfgVehicles" >> typeOf _vehicle >> "Components" >> "TransportPylonsComponent" >> "Pylons", "isClass _x"];
+    if (_pylonIndex >= (count _pyMags)) exitWith {ERROR("out of bounds"); []};
```

Comment:
no need to return `[]` inside this if-else because it exits to big if-else.
IMO it's better to remove big if-else above (`if (!(_returnValue isEqualTo []))`) and use `exitWith` there.

Reply:
This is nicer for debugging due to TRACE in the if-then.